docs/process-message/process-message.puml

```diff
@@ -0,0 +1,184 @@
+@startuml
+
+queue "message-created" as CreatedMessage
+participant ProcessMessage
+database "servicesfn1sdt" as InternalStorage
+collections profiles
+collections "message-status" as messageStatus
+collections messages
+collections "service-preferences" as servicePreferences
+collections activations
+database "message-content" as messageStorage
+
+queue "message-procesed" as ProcessedMessage
+
+CreatedMessage -> ProcessMessage : Read an item
+
+ProcessMessage -> ProcessMessage : Json Parse input from the created-message queue
+
+alt "Invalid JSON"
+  ProcessMessage -[#red]>X CreatedMessage: Throw an error
+end
+
+ProcessMessage -> ProcessMessage : Validate the shape of the item
+
+alt "Invalid CreatedMessageEvent"
+  ProcessMessage -[#red]>X CreatedMessage: Throw an error
+end
+
+activate ProcessMessage
+ProcessMessage -> InternalStorage : Ask for a CommonMessageData
+
+activate InternalStorage
+alt "Error from storage"
+  InternalStorage -[#red]> ProcessMessage : Storage error
+  ProcessMessage -[#red]>X CreatedMessage: Throw an error
+else "Data not found"
+  InternalStorage -> ProcessMessage : None
+  ProcessMessage -[#red]>X CreatedMessage: Throw an error
+else "Data does not respect the shape CommonMessageData"
+  InternalStorage -[#red]> ProcessMessage : Decode error
+  ProcessMessage -[#red]>X CreatedMessage: Throw an error
+else "Storage returns data"
+  InternalStorage -[#green]> ProcessMessage : Data
+  deactivate InternalStorage
+end
```

Comment:
When we model error handling, we don't have to list all error cases. I'd rather focus on the logic behind these checks.

For example, in this case the `core info` is: `if CoreMessageData is not found or the retrieval fails then ProcessMessage throws`

Reply:
Take a look here https://github.com/pagopa/io-messages/pull/18/commits/47bd1ae1ea9d402b7c009b53f0ab9867174ec96a